### ISSUE TITLE
connection.query does not return

### ISSUE THREAD

**Issue by icarus31** [nuintun/node-adodb#47]
Hi,

I developed an Angular 4+Electron application with node-adodb. It works perfectly in development mode, but when packaging everything (create the .exe and other electron files), the connection.query does not go to the 'then()' or the 'catch()'!

Using node-adodb version 4.0.6
Angular 4.3.6
electron 1.7.9
electron-packager: 10.1.1

main.js code

```
const ADODB = require('node-adodb');
const dbProvider = 'Provider=Microsoft.Jet.OLEDB.4.0;Data Source=';
const databasePath = path.normalize('C:\\temp\\mydb.mdb');

ipcMain.on('select', (event, type, sql) => {

    const reply = 'reply-select-' + type;

    connection = ADODB.open(dbProvider + databasePath + ";");

    connection
      .query(sql)
      .then((data) => {
        event.sender.send(reply, { status: 0, message: '', data: data });
      })
      .catch((error) => {
        console.error('ERROR with select: ' + error);
        event.sender.send(reply, { status: 1, message: error.message, data: {} });
      });
  });
```

I know it enters in the ipcMain.on, execute the ADODB.open, but after, nothing.

Any clue why or some things that you could suggest me to try?

Thanks

**Comment by nuintun** [comment]
@icarus31 See #36

**Comment by icarus31** [comment]
@nuintun, the solution is to use the 'hazardous' library instead, right? Have you tried it too? Would you mind to help me out how to do that? No too sure to understand. Thanks

**Comment by AdrianCuartas** [comment]
Same problem here. @icarus31 Did you find a way to fix it?
Thanks in advance.

**Comment by icarus31** [comment]
Hi @AdrianCuartas, nothing new from my side :-(

**Comment by nuintun** [comment]
@AdrianCuartas @icarus31 Please upload a demo~

**Comment by AdrianCuartas** [comment]
I'm sorry, I changed it for another module.
But I think the problem could be some dependency of a folder (if it exists). 

When you pack an electron application, you are not longer able to read the content of a folder inside node_modules. That's because all is packed in a single file (.asar).
You can read and import a file, of course, but not to read (fs.readdir)  the content of a folder.

**Comment by AlfiGit** [comment]
Hello, I found an issue when developing with adodb. I have created an access database (.accdb) and connected it using adodb. It seems that I can CREATE, INSERT and anything that basically **writes** to the databse, but I cannot seem to **read** anything. (SELECT doesn't work). This is my code:

`var express = require('express')
var app = express()
var fs = require('fs')
var adodb = require('node-adodb')
adodb.debug = true

var connection = adodb.open('Provider=Microsoft.ACE.OLEDB.12.0;Data Source=' + __dirname + '/WorkDB.accdb;Persist Security Info=False;')
  
app.get('/', async (request, response) => {
  var query = connection.execute("SELECT * FROM Employees")
  var data = await query;
  console.log("Data:", data)
  response.send(JSON.stringify(data))
})

app.listen(8080, err => {
  if(err) console.error(err)
  else console.log("Server listening on port 8080")
})`

Details: 

- node-adodb v4.0.9
- Windows 7 x84
- ms Access 10 (ACE.OLEDB.12.0)
